src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -133,6 +134,9 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
         }
     }
 
+    // FIXME: Could be a problem in case of dynamic shapes
+    ShapeOfToConst(model);
+
```

Comment:
Probably it is better to move the below `GraphRewrite` and add this pass first there (and then add others under the same condition)

Reply:
Done